src/Service/Telemetry.php

```diff
@@ -31,4 +39,49 @@ public function isInstanceIdPresent(): bool
     {
         return \file_exists($this->instanceIdFile);
     }
+
+    public function instanceId(): string
+    {
+        return (string) \file_get_contents($this->instanceIdFile);
+    }
+
+    public function collectAndSend(\DateTimeImmutable $date): void
+    {
+        $this->endpoint->send(
+            $this->userAgent(),
+            new Entry(
+                $date,
+                $this->instanceId(),
+                Kernel::REPMAN_VERSION,
+                $this->osVersion(),
+                $this->phpVersion(),
+                $this->query->allOrganizationsCount($date),
+                $this->query->publicOrganizationsCount($date),
+                $this->query->allPackagesCount($date),
+                $this->query->allPackagesInstalls($date),
+                $this->query->allTokensCount($date),
+                $this->query->allUsersCount($date),
+            )
+        );
+    }
+
+    private function userAgent(): string
+    {
+        return sprintf(
+            'Repman/%s (%s; %s)',
```

Comment:
this logic is in two places already (including these), maybe it would be worth creating one class for it?